__tests__/integration.js

```diff
@@ -0,0 +1,391 @@
+const awsServerlessExpress = require('../index')
+const apiGatewayEvent = require('../example/api-gateway-event.json')
+const app = require('../example/app')
+
+const server = awsServerlessExpress.createServer(app)
+const lambdaFunction = {
+  handler: (event, context) => awsServerlessExpress.proxy(server, event, context)
+}
+
+function clone(json) {
+  return JSON.parse(JSON.stringify(json))
+}
+
+function makeEvent(eventOverrides) {
+  const baseEvent = clone(apiGatewayEvent)
+  const headers = Object.assign({}, baseEvent.headers, eventOverrides.headers)
+  delete eventOverrides.headers
+  const root = Object.assign({}, baseEvent, eventOverrides)
+  root.headers = headers
```

Comment:
This headers stuff could probably be cleaned up, but since`Object.assign` does not do a deep merge, we need to first merge the `eventOverride.headers` with the base/example event headers then set those in the resulting merged event.